src/test/java/org/folio/rest/impl/base/AbstractAPITest.java

```diff
@@ -0,0 +1,317 @@
+      event -> {
+        if (event.failed()) {
+          logger.error(event.cause());
+          context.fail(event.cause());
+        } else {
+          async.complete();
+        }
+      });
+  }
+
+  /**
+   * Get emails by status
+   */
+  protected Response getEmails(String status) {
+    String okapiUrl = String.format(OKAPI_URL_TEMPLATE, userMockServer.port());
+    return RestAssured.given()
+      .port(port)
+      .contentType(MediaType.APPLICATION_JSON)
+      .header(new Header(OKAPI_HEADER_TENANT, OKAPI_TENANT))
+      .header(new Header(OKAPI_URL_HEADER, okapiUrl))
+      .header(new Header(OKAPI_HEADER_TOKEN, OKAPI_TOKEN))
+      .when()
+      .get(String.format(REST_PATH_WITH_QUERY, REST_PATH_GET_EMAILS, status, DEFAULT_LIMIT));
+  }
+
+  /**
+   * Send email notifications
+   */
+  protected Response sendEmail(EmailEntity emailEntity) {
+    String okapiUrl = String.format(OKAPI_URL_TEMPLATE, userMockServer.port());
+    return RestAssured.given()
+      .port(port)
+      .contentType(MediaType.APPLICATION_JSON)
+      .header(new Header(OKAPI_HEADER_TENANT, OKAPI_TENANT))
+      .header(new Header(OKAPI_URL_HEADER, okapiUrl))
+      .header(new Header(OKAPI_HEADER_TOKEN, OKAPI_TOKEN))
+      .body(JsonObject.mapFrom(emailEntity).toString())
+      .when()
+      .post(REST_PATH_GET_EMAILS);
+  }
+
+  /**
+   * Delete email by expirationDate and status
+   */
+  protected Response deleteEmailByDateAndStatus(String expirationDate, String status) {
+    String okapiUrl = String.format(OKAPI_URL_TEMPLATE, userMockServer.port());
+    return RestAssured.given()
+      .port(port)
+      .header(new Header(OKAPI_HEADER_TENANT, OKAPI_TENANT))
+      .header(new Header(OKAPI_URL_HEADER, okapiUrl))
+      .header(new Header(OKAPI_HEADER_TOKEN, OKAPI_TOKEN))
+      .when()
+      .delete(String.format(REST_PATH_DELETE_BATCH_EMAILS, REST_DELETE_BATCH_EMAILS, expirationDate, status));
+  }
+
+  /**
+   * Check that after sending the email successfully, it contains the email address of the recipient
+   */
+  protected void checkResponseMessage(Response response, String recipient) {
+    String responseMessage = response.getBody().asString();
+    assertEquals(String.format(SUCCESS_SEND_EMAIL, recipient), responseMessage);
+  }
+
+  /**
+   * Find the sent email message on the SMTP server by the sender
+   */
+  protected WiserMessage findMessageOnWiserServer(String sender) {
+    return wiser.getMessages().stream()
+      .filter(msg -> {
+        try {
+          return isContainsSenderAddress(sender, msg.getMimeMessage().getFrom());
+        } catch (MessagingException ex) {
+          logger.debug(ex);
+          throw throwAssertionFailedError(sender);
+        }
+      })
+      .findFirst()
+      .orElseThrow(() -> throwAssertionFailedError(sender));
+  }
+
+  private AssertionFailedError throwAssertionFailedError(String sender) {
```

Comment:
method's name does not reflect what it does correctly. it creates new error but does not throw it

Reply:
Updated